### PR TITLE
637 cache key vault values

### PR DESCRIPTION
**Pull request by chenrujun**
## Summary
Fix issue #637 

## Issue Type
- Bug fixing

## Additional Information

### CODE REVIEW

File: azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultPropertySource.java [hunk 1]
```diff
@@ -19,7 +19,7 @@ public KeyVaultPropertySource(KeyVaultOperation operation) {
 
 
     public String[] getPropertyNames() {
```
Comment (saragluna) [inline azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultPropertySource.java:21]: This method will not be called by Spring now, right?

File: azure-spring-boot-samples/azure-keyvault-secrets-spring-boot-sample/pom.xml [hunk 2]
```diff
@@ -23,10 +23,27 @@
...
+                <version>${spring-cloud.version}</version>
+                <type>pom</type>
+                <scope>import</scope>
+            </dependency>
+        </dependencies>
+    </dependencyManagement>
+
     <properties>
         <project.rootdir>${project.basedir}/../..</project.rootdir>
+        <spring-cloud.version>Greenwich.SR3</spring-cloud.version>
```
[the remark this answers is not in the document]
Comment (chenrujun) [reply azure-spring-boot-samples/azure-keyvault-secrets-spring-boot-sample/pom.xml:46]: IMU, sample project have 2 purpose:
1. Lest user know how to use it. It should as simple as possible.
2. Test the functions. more test case is better.

After your comment, I think `1` is more important, I'll remove the dependency.

File: azure-spring-boot-samples/azure-keyvault-secrets-spring-boot-sample/src/main/resources/application.properties [hunk 3]
```diff
@@ -2,3 +2,7 @@ azure.keyvault.client-id=put-your-azure-client-id-here
 azure.keyvault.client-key=put-your-azure-client-key-here
 azure.keyvault.uri=put-your-azure-keyvault-uri-here
 azure.keyvault.tenant-id=
+
+property.in.raw=property.in.raw.value
```
[the remark this answers is not in the document]
Comment (chenrujun) [reply azure-spring-boot-samples/azure-keyvault-secrets-spring-boot-sample/src/main/resources/application.properties:6]: To make the sample simple, I deleted it.

File: azure-spring-boot-samples/azure-keyvault-secrets-spring-boot-sample/src/main/java/sample/keyvault/SampleApplication.java [hunk 4]
```diff
@@ -6,28 +6,41 @@
...
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 
+@Slf4j
 @SpringBootApplication
 public class SampleApplication implements CommandLineRunner {
-    private static final Logger LOGGER = LoggerFactory.getLogger(SampleApplication.class);
 
-    @Value("${yourSecretPropertyName}")
-    private String mySecretProperty;
+    @Value("${yourSecretPropertyName9}")
```
[the remark this answers is not in the document]
Comment (chenrujun) [reply azure-spring-boot-samples/azure-keyvault-secrets-spring-boot-sample/src/main/java/sample/keyvault/SampleApplication.java:19]: Done.

File: azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultPropertySource.java (hunk 1 above)
Comment (chenrujun) [reply azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultPropertySource.java:21]: Yes.
Spring doesn't know we have this method.
Now this method is just used in test case.
I think it's OK to keep it, maybe we need it in future.

**Comment by chenrujun** [comment]
@saragluna Build passed. Please merge it. 🙏

**Pull request merged**